tests/common/test_cgroupconfigurator.py

```diff
@@ -17,117 +17,85 @@
-from azurelinuxagent.common.utils import fileutil
-from tests.utils.cgroups_tools import CGroupsTools
+from tests.common.mock_cgroup_commands import mock_cgroup_commands
 from tests.tools import AgentTestCase, patch
 
 
-class CGroupConfiguratorTestCase(AgentTestCase):
-    @classmethod
-    def setUpClass(cls):
-        AgentTestCase.setUpClass()
-
-        # Use the file system implementation of CGroupsApi (FileSystemCgroupsApi)
-        cls.mock_is_systemd = patch("azurelinuxagent.common.cgroupapi.CGroupsApi._is_systemd", return_value=False)
-        cls.mock_is_systemd.start()
-
-        # Use the default implementation of osutil
-        cls.mock_get_osutil = patch("azurelinuxagent.common.cgroupconfigurator.get_osutil", return_value=DefaultOSUtil())
-        cls.mock_get_osutil.start()
-
-        # Currently osutil.is_cgroups_supported() returns False on Travis runs. We need to revisit this design; in the
-        # meanwhile mock the method to return True
-        cls.mock_is_cgroups_supported = patch("azurelinuxagent.common.osutil.default.DefaultOSUtil.is_cgroups_supported", return_value=True)
-        cls.mock_is_cgroups_supported.start()
-
-        # Mounting the cgroup filesystem requires root privileges. Since these tests do not perform any actual operation on cgroups, make it a noop.
-        cls.mock_mount_cgroups = patch("azurelinuxagent.common.osutil.default.DefaultOSUtil.mount_cgroups")
-        cls.mock_mount_cgroups.start()
-
+class CGroupConfiguratorSystemdTestCase(AgentTestCase):
     @classmethod
     def tearDownClass(cls):
-        cls.mock_mount_cgroups.stop()
-        cls.mock_is_cgroups_supported.stop()
-        cls.mock_get_osutil.stop()
-        cls.mock_is_systemd.stop()
-
+        CGroupConfigurator._instance = None
         AgentTestCase.tearDownClass()
 
-    def setUp(self):
-        AgentTestCase.setUp(self)
-        CGroupConfigurator._instance = None  # force get_instance() to create a new instance for each test
-
-        self.cgroups_file_system_root = os.path.join(self.tmp_dir, "cgroup")
-        os.mkdir(self.cgroups_file_system_root)
-        os.mkdir(os.path.join(self.cgroups_file_system_root, "cpu"))
-        os.mkdir(os.path.join(self.cgroups_file_system_root, "memory"))
-
-        self.mock_cgroups_file_system_root = patch("azurelinuxagent.common.cgroupapi.CGROUPS_FILE_SYSTEM_ROOT", self.cgroups_file_system_root)
-        self.mock_cgroups_file_system_root.start()
-
-    def tearDown(self):
-        self.mock_cgroups_file_system_root.stop()
+    @staticmethod
+    def _get_new_cgroup_configurator_instance(initialize=True):
+        CGroupConfigurator._instance = None
+        configurator = CGroupConfigurator.get_instance()
+        if initialize:
+            with patch('azurelinuxagent.common.cgroupapi.CGroupsApi.cgroups_supported', return_value=True):
+                with patch('azurelinuxagent.common.cgroupapi.CGroupsApi._is_systemd', return_value=True):
```

Comment:
Before this PR, we were forcing FileSystemCgroupsApi here. Any reason for the change? I forget why filesystem was favoured before.

Reply:
Yes, with these changes FileSystemCgroupsApi is dead code. I do not want to delete it yet since we will eventually get to distros without systemd but for now all the code is going with the systemd path. I updated these tests to follow the same